File: Basketball2.java
// Anthony Dike started: 3/3/18
//
// This is VERSION 2 of a program that
// will simulate basketball games like NBA 2k.
//
// In this version I will add a welcome and closing message
// & add a feature that adds an overtime to the game
// if the score is tied after the regulation time.

import java.util.Random;
import java.util.Scanner;

/*
Steps:
1) split code labor into methods
2) OvertimeTeam1 & 2 - random added int amount to score
3) Game - add if else for overtime trigger 
4) Main - print welcome & closing messages & if yes play else if no break
*/

public class Basketball2 {
	// global variables
	public static int Team1;
	public static int Team2;

	public static int OvertimeTeam1(){
		Random rand = new Random();
		//Overtime increases maxScore # of points by 30
		int overtimeAdder = rand.nextInt(30);
		int team1OT = Team1 + overtimeAdder;
		return team1OT;

	}
	public static int OvertimeTeam2(){
		Random rand = new Random();
		//Overtime increases maxScore # of points by 30
		int overtimeAdder = rand.nextInt(30);
		int team2OT = Team2 + overtimeAdder;
		return team2OT;

	}
	public static void Game(){
		// initialize
		int maxScore = 122;
		int minScore = 98;

		// random class
		Random rand = new Random();

		// assign ints to team variables
		Team1 = rand.nextInt(maxScore - minScore) + minScore;
		Team2 = rand.nextInt(maxScore - minScore) + minScore;

		// convert int to string
		String team1Value = Integer.toString(Team1);
		String team2Value = Integer.toString(Team2);
		String overtimeTeam1 = Integer.toString(OvertimeTeam1());
		String overtimeTeam2 = Integer.toString(OvertimeTeam2());

		if (Team1 == Team2){
			System.out.println("There was an overtime!");
			System.out.println("REGULATION: Team1 - " + team1Value + 
				" Team2 - " + team2Value);
			//prints overtime
			System.out.println("OVERTIME: Team 1 - " + overtimeTeam1 + 
				" Team 2 - " + overtimeTeam2);
		} else {
			System.out.println("SCORE: Team1 - " + team1Value + " Team2 - " + team2Value);
		}

	}
	
	public static void main(String[] arguments){
		// welcome message
		System.out.println();
		System.out.println("----------------------------------------------");
		System.out.println("| HELLO, WELCOME TO THE BASKETBALL SIMULATOR |");
		System.out.println("----------------------------------------------");
		System.out.println();

		Scanner in = new Scanner(System.in);

		boolean proceed = false;

		while(proceed == false){

			System.out.println("\t Would you like to start a game?");
    		System.out.println();
    		System.out.print("(YES or NO): ");
    		String answer = in.nextLine();
    		System.out.println();

    		if (answer.equals("yes")){
    			System.out.println();
    			Game();
    			System.out.println();

    		} else if (answer.equals("no")){
    			proceed = true;
    			break;

    		} else {
    			System.out.println("INVALID INPUT. Try Again.");
    			System.out.println();
    		}
		}

		// closing message
		System.out.println();
		System.out.println("----------------------------------------------");
		System.out.println("|       GOODBYE, THANK YOU FOR PLAYING!      |");
		System.out.println("----------------------------------------------");
		System.out.println();
	}



}

File: Basketball2.py
## Anthony Dike started: 3/3/18
##
## This is VERSION 2 of a program that
## will simulate basketball games like NBA 2k.
##
## In this version I will add a welcome and closing message
## & add a feature that adds an overtime to the game
## if the score is tied after the regulation time.

import random

'''
Steps:
1) print a welcome message
2) modulate using functions
3) add if-else statement that makes a tie continue to an extra round
4) increase max score and allow random int to be added to tie
5) print regulation score and then if tie then print overtime score


'''

def OvertimeTeam1():
    overtimeAdder = random.randint(0,30)
    global Team1
    Team1 += overtimeAdder
    global team1OT
    team1OT = Team1
    return team1OT
    

def OvertimeTeam2():
    overtimeAdder = random.randint(0,30)
    global Team2
    Team2 += overtimeAdder
    global team2OT
    team2OT = Team2
    return team2OT
    


def Game():
    
    # create max and min scores
    maxScore = 122
    minScore = 98

    # initialize team variables
    global Team1
    Team1 = random.randint(minScore, maxScore)
    global Team2
    Team2 = random.randint(minScore, maxScore)

    # if tie trigger overtime
    if Team1 == Team2:
        print("There was an overtime!")
        print("REGULATION: Team1 - " + str(Team1) + " Team 2 - " + str(Team2))
        # prints overtime
        print("OVERTIME: Team 1 - " + str(OvertimeTeam1()) +
              " Team 2 - " + str(OvertimeTeam2()))
        
    else:
        print("SCORE: Team1 - " + str(Team1) + " Team 2 - " + str(Team2))
        


# welcome message
print()
print("----------------------------------------------")
print("| HELLO, WELCOME TO THE BASKETBALL SIMULATOR |")
print("----------------------------------------------")
print()


proceed = False

while proceed == False:
    print("\t Would you like to start a game?")
    print()
    answer = input("(YES or NO): ")
    print()
    
    if answer.lower() == "yes":
        print()
        Game()
        print()
    elif answer.lower() == "no":
        proceed = True
        break
    else:
        print("INVALID INPUT. Try Again.")
        print()

# closing message
print()
print("----------------------------------------------")
print("|       GOODBYE, THANK YOU FOR PLAYING!      |")
print("----------------------------------------------")
print()





